sdk/keyvault/keyvault-certificates/test/identifier.test.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as assert from "assert";
+import { KeyVaultCertificatesIdentifier } from "../src";
+
+describe("KeyVault Certificates Identifier", () => {
+  it("It should work with a URI of a certificate before it gets a version", async function() {
+    const uri = "https://keyvault-name.vault.azure.net/certificates/certificate-name/pending";
+    const identifier = new KeyVaultCertificatesIdentifier(uri);
+
+    // Assert doesn't have deepEqual on the browser.
+    assert.equal(
+      JSON.stringify(identifier),
+      JSON.stringify({
+        collection: "certificates",
+        id: "https://keyvault-name.vault.azure.net/certificates/certificate-name/pending",
+        vaultUrl: "https://keyvault-name.vault.azure.net",
+        version: "pending",
+        name: "certificate-name"
+      })
+    );
+  });
+
+  it("It should work with a URI of a certificate with a specific version", async function() {
+    const uri = "https://keyvault-name.vault.azure.net/certificates/certificate-name/version";
+    const identifier = new KeyVaultCertificatesIdentifier(uri);
+
+    // Assert doesn't have deepEqual on the browser.
+    assert.equal(
+      JSON.stringify(identifier),
+      JSON.stringify({
+        collection: "certificates",
+        id: "https://keyvault-name.vault.azure.net/certificates/certificate-name/version",
+        vaultUrl: "https://keyvault-name.vault.azure.net",
+        version: "version",
+        name: "certificate-name"
+      })
+    );
+  });
+
+  it("It should work with a URI of a certificate's issuer", async function() {
```

Comment:
Is there value in parsing an issuer since this isn't a proper collection in the sense we mean it? Yes, it might work, but testing it in public code implies it should always work. We decided we aren't, for now, supporting certificate issuers in this way since there's no scenario we can think of currently.

Reply:
This comes from replacing the generated method with this new API in the insides of our client. Besides the three collections I was aware of, I saw that the client was also parsing URIs of issuers. The value is just documenting the behaviors we expect to be currently working, also to make sure they don't suddenly break later.

Reply:
Was it not tested at all before? Or could you still use the base class from which you're deriving to test these specific cases?

Reply:
Inside of the constructor of the new Identifiers, I am using the same core function that worked to parse identifiers that we already use from the generated code. Now I'm using this new identifier instead of the core generated function inside of our client's source code, which helped me log out and see what were all of the URIs that this function would receive from all of our live tests. Based on that, besides keys, secrets, and certificates, I was only able to see a different case for the certificate issues. Since it's a different case, even in the structure of the URI, I decided to write a test for it (the similar tests for keys and secrets were a copy-paste mistake and I have removed them).

Before this PR, we didn't have unit tests for this because this was just the generated code. We don't test the generated code. I'm now explicitly testing this since this is going to be a new part of our public API, if that makes sense.